### ISSUE TITLE
Whitelist comments by keyword (reverse no-warning-comments)

### ISSUE THREAD

**Issue by ViggoV** [eslint/eslint#11590]
**Please describe what the rule should do:**
It should only allow comments prepended with one of a set of keywords, like `no-warning-comments` but whitelisting rather than blacklisting. Examples could be `['note', 'info', 'example']`, so loose comments like outcommented code would throw an error/warning.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[x] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
// note: n must be an integer for code to work <- no warning
n = Math.floor(n);

// console.log('this was for debugging') <- warning

/* example: <- no warning
const x = SomeFunction(someInput);
x.doSomeStuff();
*/

/* <- warning
for (let i; i < 10; i++) {
  doStuffThatDidntWorkProperly();
}
*/
```

**Why should this rule be included in ESLint (instead of a plugin)?**
It is a basic rule that makes a lot of sense, at least to me. Especially since comment code rules are requested frequently and this could solve that issue in many cases.

**Are you willing to submit a pull request to implement this rule?**
WIlling, but not sure if I'm able
